workshops/test/test_util.py

```diff
@@ -376,195 +376,326 @@ def test_attendance_increases(self):
-    def test_wrong_url(self):
-        with self.assertRaises(requests.exceptions.HTTPError):
-            url = 'http://test.github.io/2015-07-13-test/'
-            parse_tags_from_event_index(url)
-
-    @patch.object(requests, 'get')
-    def test_parsing_event_index(self, mock_get):
-        mock_get.return_value.status_code = 200
-        mock_get.return_value.text = """---
+    def test_generating_url_to_index(self):
+        tests = [
+            'http://swcarpentry.github.io/workshop-template',
+            'https://swcarpentry.github.com/workshop-template',
+            'http://swcarpentry.github.com/workshop-template/',
+        ]
+        expected = ('https://raw.githubusercontent.com/swcarpentry/'
+                    'workshop-template/gh-pages/index.html')
+        for url in tests:
+            with self.subTest(url=url):
+                self.assertEqual(expected, generate_url_to_event_index(url))
+
+    def test_finding_tags_on_index(self):
+        content = """---
 layout: workshop
 root: .
 venue: Euphoric State University
 address: Highway to Heaven 42, Academipolis
-country: USA
+country: us
 language: us
 latlng: 36.998977, -109.045173
 humandate: Jul 13-14, 2015
 humantime: 9:00 - 17:00
 startdate: 2015-07-13
-enddate: 2015-07-14
-instructor: ["Hermione Granger", "Harry Potter", "Ron Weasley",]
+enddate: "2015-07-14"
+instructor: ["Hermione Granger", "Ron Weasley",]
 helper: ["Peter Parker", "Tony Stark", "Natasha Romanova",]
-contact: hermione@granger.co.uk, rweasley@ministry.gov.uk
+contact: hermione@granger.co.uk, rweasley@ministry.gov
 etherpad:
 eventbrite: 10000000
----
+----
+Other content.
+"""
+        expected = {
+            'startdate': '2015-07-13',
+            'enddate': '2015-07-14',
+            'country': 'us',
+            'venue': 'Euphoric State University',
+            'address': 'Highway to Heaven 42, Academipolis',
+            'latlng': '36.998977, -109.045173',
+            'language': 'us',
+            'instructor': 'Hermione Granger, Ron Weasley',
+            'helper': 'Peter Parker, Tony Stark, Natasha Romanova',
+            'contact': 'hermione@granger.co.uk, rweasley@ministry.gov',
+            'eventbrite': '10000000',
+        }
+        self.assertEqual(expected, find_tags_on_event_index(content))
+
+    def test_finding_tags_on_website(self):
+        content = """
+<html><head>
+<meta name="slug" content="2015-07-13-test" />
+<meta name="startdate" content="2015-07-13" />
+<meta name="enddate" content="2015-07-14" />
+<meta name="country" content="us" />
+<meta name="venue" content="Euphoric State University" />
+<meta name="address" content="Highway to Heaven 42, Academipolis" />
+<meta name="latlng" content="36.998977, -109.045173" />
+<meta name="language" content="us" />
+<meta name="invalid" content="invalid" />
+<meta name="instructor" content="Hermione Granger, Ron Weasley" />
+<meta name="helper" content="Peter Parker, Tony Stark, Natasha Romanova" />
```

Comment:
These should be [pipe-delimited now][1], no?

[1]: https://github.com/swcarpentry/workshop-template/commit/1e19a53c2fa00fe1becacbab916cc70d25262b3f